### PR TITLE
[UTOPIA-1395] Selected program area roles not visible when partially signed in Final Review

### PR DESCRIPTION
**Pull request by JieunSon96**
# Description

This PR includes the following proposed change(s):

- [UTOPIA-1395](https://apps.itsm.gov.bc.ca/jira/browse/UTOPIA-1395)
- Fixed ambiguity in the logic that determines whether to display the view or edit section in each section
- Changed the function name: `allowUserReviewProgramArea` -> `viewUserReviewProgramArea` 

 | viewUserReviewProgramArea | isAcknowledged | Section |
|---------------------------|----------------|---------|
| false                     | false          | edit    |
| false                     | true           | view    |
| true                      | false          | view    |
| true                      | true           | view    |

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactoring / Documentation
- [ ] Updating Testing Framework(s)
- [ ] Version change

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

## Development Dependency Working Agreement
- [x] My code DOES NOT include the importing of new dependencies into the DPIA ecosystem
- [ ] My code DOES include the importing of new dependencies into the DPIA ecosystem
**If new dependencies are being introduced to the DPIA ecosystem:**
- [ ] The functionality of the dependency drastically reduces code complexity and makes my changes more easily maintainable and readable 
- [ ] The dependency being introduced does not contain multiple layers of nested dependencies introducing maintainability complexity to the DPIA ecosystem

## Frontend Development Changes
- [ ] N/A
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to project documentation or diagrams that reflect my changes
- [ ] New and existing unit tests pass locally with my changes
- [ ] My code follows Airbnb React Style Guidelines

## API Development Changes
- [x] N/A
- [ ] I have performed a self-review of my own code
- [ ] My code follows standards and practices outlined in the BC Government API Development Guidelines
- [ ] New and existing unit tests pass locally with my changes
- [ ] My changes includes Swagger documentation updates that reflect the changes I am introducing

## Definition of Done

Definition of Done

### CODE REVIEW

**Comment by NoorChasib** [comment]
so simple yet so effective

**Review by NoorChasib — DISMISSED** [review]

File: src/frontend/src/components/public/PIAFormTabs/review/ProgamArea/displayProgramArea.tsx [hunk 1]
```diff
@@ -37,8 +37,11 @@ const DisplayProgramArea = (props: IDisplayProgramAreaProps) => {
   const canEditProgramAreaReview =
     reviewPagePrivilegeParams?.editProgramAreaReview ?? false;
 
-  const allowUserReviewProgramArea = () => {
-    if (!canEditProgramAreaReview) return false;
+  // Logic Explanation:
+  //   - When this function returns true, it means the readonly.
+  //   - When this function returns false, it means allow editing the review.
+  const viewUserReviewProgramArea = () => {
+    if (canEditProgramAreaReview) return false;
```
Comment (BradyMitch) [inline src/frontend/src/components/public/PIAFormTabs/review/ProgamArea/displayProgramArea.tsx:44]:
```suggestion
    const reviewerRoleAcknowledged = Object(props.pia?.review?.programArea)
      ?.reviews?.[reviewerRole]?.isAcknowledged;
    if (canEditProgramAreaReview || !reviewerRoleAcknowledged) return false;
```

File: src/frontend/src/components/public/PIAFormTabs/review/ProgamArea/displayProgramArea.tsx [hunk 2]
```diff
@@ -37,8 +37,11 @@ const DisplayProgramArea = (props: IDisplayProgramAreaProps) => {
   const canEditProgramAreaReview =
     reviewPagePrivilegeParams?.editProgramAreaReview ?? false;
 
-  const allowUserReviewProgramArea = () => {
-    if (!canEditProgramAreaReview) return false;
+  // Logic Explanation:
+  //   - When this function returns true, it means the readonly.
+  //   - When this function returns false, it means allow editing the review.
+  const viewUserReviewProgramArea = () => {
```
Comment (BradyMitch) [inline src/frontend/src/components/public/PIAFormTabs/review/ProgamArea/displayProgramArea.tsx:43]:
```suggestion
  const viewMode = (reviewerRole: string) => {
```

Comment (BradyMitch) [reply src/frontend/src/components/public/PIAFormTabs/review/ProgamArea/displayProgramArea.tsx:43]: along with changes down below to the function call

Comment (BradyMitch) [reply src/frontend/src/components/public/PIAFormTabs/review/ProgamArea/displayProgramArea.tsx:43]: or call the function `readOnly`

File: src/frontend/src/components/public/PIAFormTabs/review/ProgamArea/displayProgramArea.tsx [hunk 3]
```diff
@@ -37,8 +37,11 @@ const DisplayProgramArea = (props: IDisplayProgramAreaProps) => {
   const canEditProgramAreaReview =
     reviewPagePrivilegeParams?.editProgramAreaReview ?? false;
 
-  const allowUserReviewProgramArea = () => {
-    if (!canEditProgramAreaReview) return false;
+  // Logic Explanation:
+  //   - When this function returns true, it means the readonly.
+  //   - When this function returns false, it means allow editing the review.
```
Comment (antsand) [inline src/frontend/src/components/public/PIAFormTabs/review/ProgamArea/displayProgramArea.tsx:42]: i may add to the comments if you are drafter, etc you will always be in view mode
if you are a reviewer, you can only edit your review etc. 

The processing of what make it true or false can be further elaborated.

File: src/frontend/src/components/public/PIAFormTabs/review/ProgamArea/displayProgramArea.tsx (hunk 2 above)
Comment (JieunSon96) [reply src/frontend/src/components/public/PIAFormTabs/review/ProgamArea/displayProgramArea.tsx:43]: thanks

> along with changes down below to the function call

thanks, I think the viewMode function name would be more intuitive

File: src/frontend/src/components/public/PIAFormTabs/review/ProgamArea/displayProgramArea.tsx (hunk 3 above)
Comment (BradyMitch) [reply src/frontend/src/components/public/PIAFormTabs/review/ProgamArea/displayProgramArea.tsx:42]: If you are a drafter you can add a review/edit your own review

Comment (JieunSon96) [reply src/frontend/src/components/public/PIAFormTabs/review/ProgamArea/displayProgramArea.tsx:42]: > i may add to the comments if you are drafter, etc you will always be in view mode if you are a reviewer, you can only edit your review etc.
> 
> The processing of what make it true or false can be further elaborated.

yes right, I will add more elaborated comments in terms of the drafter, reviewer etc.

File: src/frontend/src/components/public/PIAFormTabs/review/ProgamArea/displayProgramArea.tsx [hunk 4]
```diff
@@ -110,7 +115,7 @@ const DisplayProgramArea = (props: IDisplayProgramAreaProps) => {
             </div>
           ) : (
             <div className="d-flex align-items-center" key={index}>
-              {!allowUserReviewProgramArea() ||
+              {viewMode(role) ||
               Object(props.pia?.review?.programArea)?.reviews?.[role]
```
Comment (BradyMitch) [inline src/frontend/src/components/public/PIAFormTabs/review/ProgamArea/displayProgramArea.tsx:119]: The part after viewMode is not required anymore

**Review by BradyMitch — APPROVED** [review]

**Pull request merged**